CMakeLists.txt

```diff
@@ -25,7 +25,7 @@
 
 # temporary
 set(CMAKE_WARN_DEPRECATED OFF)
-add_compile_options(-Wno-format-extra-args)
+add_compile_options(-Wno-format-extra-args -mf16c)
```

Comment:
Is this safe to add unconditionally? Isn't this enabling the F16C ISA extension? What if the CPU doesn't support it?

Reply:
> Isn't this enabling the F16C ISA extension?

if I understand correctly, yes. 

> What if the CPU doesn't support it?

based on this list https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html we'd be hard pressed to find a modern CPU that doesn't support it.

Reply:
> based on this list https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html we'd be hard pressed to find a modern CPU that doesn't support it.

I don't think we want to reduce the portability on "less-modern" CPUs or, say Arm hosts, just for this? Could we check for it?

Reply:
I'll look into if we can check but let's say this option is not supported, then what?

Reply:
@pjaaskel I'll merge this and open an issue to implement checks/whatever we decide. Not being able to compile main on JLSE is not good right now.